lib/webResource.ts

```diff
@@ -298,65 +298,90 @@ export class WebResource {
+   * Example:
+   *    - query-parameter-value in "object" format: { "query-parameter-name": { value: "query-parameter-value", skipUrlEncoding: true } }
+   *    - query-parameter-value in "string" format: { "query-parameter-name": "query-parameter-value"}.
+   * Note: "If options.url already has some query parameters, then the value provided in options.queryParameters will be appended to the url.
+   */
   queryParameters?: { [key: string]: any | ParameterValue };
+  /**
+   * The path template of the request url. Either provide the "url" or provide the "pathTemplate" in
+   * the options object. Both the options are mutually exclusive.
+   * Example: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}"
+   */
   pathTemplate?: string;
+  /**
+   * The base url of the request. Default value is: "https://management.azure.com". This is
+   * applicable only with pathTemplate. If you are providing options.url then it is expected that
+   * you provide the complete url.
+   */
   baseUrl?: string;
+  /**
+   * A dictionary of path parameters that need to be replaced with actual values in the pathTemplate.
+   * Here the key is the "path-parameter-name" and the value is the "path-parameter-value".
+   * The "path-parameter-value" can be of type "string"  or it can be of type "object".
+   * The "object" format should be used when you want to skip url encoding. While using the object format,
+   * the object must have a property named value which provides the "path-parameter-value".
+   * Example:
+   *    - path-parameter-value in "object" format: { "path-parameter-name": { value: "path-parameter-value", skipUrlEncoding: true } }
+   *    - path-parameter-value in "string" format: { "path-parameter-name": "path-parameter-value" }.
+   */
   pathParameters?: { [key: string]: any | ParameterValue };
   formData?: { [key: string]: any };
+  /**
+   * A dictionary of request headers that need to be applied to the request.
+   * Here the key is the "header-name" and the value is the "header-value". The header-value MUST be of type string.
+   *  - ContentType must be provided with the key name as "Content-Type". Default value "application/json; charset=utf-8".
+   *  - "Transfer-Encoding" is set to "chunked" by default if "options.bodyIsStream" is set to true.
+   *  - "Content-Type" is set to "application/octet-stream" by default if "options.bodyIsStream" is set to true.
+   *  - "accept-language" by default is set to "en-US"
+   *  - "x-ms-client-request-id" by default is set to a new Guid. To not generate a guid for the request, please set options.disableClientRequestId to true
+   */
   headers?: { [key: string]: any };
+  /**
+   * When set to true, instructs the client to not set "x-ms-client-request-id" header to a new Guid().
+   */
   disableClientRequestId?: boolean;
+  /**
+   * The request body. It can be of any type. This method will JSON.stringify() the request body.
```

Comment:
Can we even succinctly describe what we will do to the body any more? Certainly no references to "this method" as there is no method being described by this doc comment..

Reply:
Done in https://github.com/Azure/ms-rest-js/pull/166/commits/46645f42238ef9e2240d7848640a294f7d324f5a.